### ISSUE TITLE
Errors aren't propagated to the parent node

### ISSUE THREAD

**Issue by Axxiss** [absinthe-graphql/absinthe#272]
Errors seems not to be propagated to the parent node

Let 's say we have the following schema

```
  object :profile do
    field :id, non_null(:id)
    field :first_name, non_null(:string)
    field :friends, list_of(:profile)
  end
```

and we do the following query

```
{
  profile(id: 1( {
    id
    firstName
    friends {
      id
      firstName
    }
  }
}
```

If `firstName` of one of friend's item is resolved to null then an exception is raised but according to graphql specification the error should be propagated to the parent node then if the parent node is optional set it to null otherwise continue going up.

http://facebook.github.io/graphql/#sec-Errors-and-Non-Nullability

> Since Non-Null type fields cannot be null, field errors are propagated to be handled by the parent field. If the parent field may be null then it resolves to null, otherwise if it is a Non-Null type, the field error is further propagated to it’s parent field.

**Comment by benwilson512** [comment]
Hey! I closed this briefly by accident sorry.

This is a known limitation. I do still think it's important to emit some kind of server side error notifying the developer of the API that this issue happened, because to me it seems to constitute a big on the part of the API.

But yea, we are being more strict than we need to be at the moment. Fixing this is on the TODO list after I finish the branch I'm currently on.

**Comment by benwilson512** [comment]
In each case that we ran across this in our own application we were better off just fixing our app to either maintain the promise of `non_null` or stop promising something we couldn't guarantee.

**Comment by Axxiss** [comment]
Thanks for the quick reply :)

> either maintain the promise of non_null or stop promising something we couldn't guarantee.

I think that makes sense and I would stick to that as much as possible.

Anyways, I just wanted to raise the issue as I didn't know if it was a know issue or not.

**Comment by benwilson512** [comment]
This was fixed in the v1.4.0 release by the way.
